include/seqan3/search/detail/search.hpp

```diff
@@ -80,41 +80,30 @@ inline auto search_single(index_t const & index, query_t & query, configuration_
     };
 
     // choose mode
-    if constexpr (search_traits_t::search_best_hits)
+    if constexpr (search_traits_t::search_best_hits ||
+                  search_traits_t::search_all_best_hits ||
+                  search_traits_t::search_strata_hits)
     {
         detail::search_param max_error2{max_error};
         max_error2.total = 0;
         while (internal_hits.empty() && max_error2.total <= max_error.total)
         {
-            detail::search_algo<true>(index, query, max_error2, internal_delegate);
+            // search_traits_t::search_best_hits:     abort_on_hit = true
```

Comment:
?

Reply:
This is pure documentation on why I set `abort_on_hit` to `!search_traits_t::search_all_best_hits`
I can remove it?

Reply:
It is ok to document the intention but I would prefer an actual sentence explaining the rational.

Reply:
I rephrased this into sentences so it does not look like outcommented code.

Reply:
The rational is that when you want to find *all* hits, you cannot stop once you found any hit.
Since this will end in recursive function calls, you can think of it as traversing a search tree, and you have to find all paths in the tree that satisfy the hit condition.
If you only want the *best* hit, you stop after finding the first hit (the hit with the least errors).
If you are in *strata* mode, you do the same as with best hits, but then do the extra stuff afterwards.

Reply:
Thanks. I changed it accordingly.